ctapipe/io/serializer.py

```diff
@@ -0,0 +1,356 @@
+"""
+Serialize ctapipe containers to file
+"""
+
+from astropy.table import Table, Column
+from ctapipe.core import Container
+from ctapipe.core import Component
+from abc import ABC, abstractmethod
+from astropy import log
+from pickle import load
+from pickle import dump
+from pickle import PickleError
+from traitlets import Unicode
+import numpy as np
+import gzip
+
+__all__ = ['FitsSerializer', 'PickleSerializer']
+log.setLevel('INFO')
+
+
+# PICKLE GZIP Implementation
+
+class Serializer(ABC, Component):
+    """
+    Base class for context manager to save Containers to file
+    """
+    def __init__(self, file,  overwrite=False):
+        """
+        Parameters
+        ----------
+        file:  Unicode
+            input or output full path file name
+        overwrite: Bool
+            overwrite outfile file if it already exists
+        """
+        self.file = file
+        self.overwrite = overwrite
+        super().__init__()
+
+    def __enter__(self):
+        log.debug("Serializing on {0}".format(self.file))
+        return self
+
+    def __exit__(self, *args):
```

Comment:
I think if a serializer keeps all objects in Memory, then `__exit__` should call write.

Also, the call signature for exit should be `__exit__(self, exc_type, exc_value, traceback)`, see https://docs.python.org/3/reference/datamodel.html#object.__exit__